hazelcast/src/main/java/com/hazelcast/jet/impl/AbstractJobProxy.java

```diff
@@ -227,7 +228,7 @@ public final JobStatus getStatus() {
 
     @Override
     public final boolean isUserCancelled() {
-        if (isLightJob()) {
+        if (isLightJob() || !isJobSuspendable(getConfig())) {
```

Comment:
this part is at least incomplete. You can get non-suspendable Job by id, then there will be no future.

Reply:
I checked it, and everything works
![image](https://github.com/hazelcast/hazelcast/assets/10107557/3ea86e22-4b37-4c17-9472-13442f90459e)

Reply:
no, it does not: https://github.com/hazelcast/hazelcast/pull/26037/commits/104e64211d771cee49f6b8e3be296f5f6308f846

BTW, I also fixed HZ-3882 but the tests for it require #26007